pkg/ebpf/dns_snooper.go

```diff
@@ -176,6 +169,36 @@ func (s *SocketFilterSnooper) pollStats() {
 	}
 }
 
+func (s *SocketFilterSnooper) start(filter *bpflib.SocketFilter) error {
+	setupErr := make(chan error)
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+
+		// Switch to root network namespace so we can snoop DNS traffic for all containers
+		runtime.LockOSThread()
+		defer runtime.UnlockOSThread()
+		rootNS, err := netns.GetFromPid(1)
+		if err != nil {
+			setupErr <- err
+			return
+		}
+		netns.Set(rootNS)
+
+		packetSrc, err := newPacketSource(filter)
+		if err != nil {
+			setupErr <- err
+			return
+		}
+
+		s.source = packetSrc
+		setupErr <- err
```

Comment:
nit: we should write this a `setupErr <- nil` to make it clearer what's happening